src/Tribe/Utils/Date_I18n.php

```diff
@@ -0,0 +1,32 @@
+<?php
+namespace Tribe\Utils;
+
+use Tribe__Date_Utils as Dates;
+use DateTime;
+use DateTimeImmutable;
+
+class Date_I18n extends DateTime {
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @return Date_I18n Localizable variation of DateTime.
+	 */
+	public static function createFromImmutable( $datetime ) {
+		return new self( $datetime->format( Dates::DBDATETIMEFORMAT ), $datetime->getTimezone() );
+	}
+
+	/**
+	 * Returns a translated string usign the params from this DateTime instance.
+	 *
+	 * @since  TBD
+	 *
+	 * @param  string $string Format to be used in the translation.
+	 *
+	 * @return string         Translated date.
+	 */
+	public function format_i18n( $string ) {
+		$unix_date  = $this->format( 'U' );
```

Comment:
Codex says the `$unix_date` is "Unix timestamp, offset by the desired timezone."
Is the timezone offset taken into account here?

Reply:
Yeah it takes in cosideration.